### PR TITLE
Use CachedOnlineStore in PreviewTrackManager

### PR DESCRIPTION
**Pull request by UselessToucan**
see https://github.com/ppy/osu-framework/pull/2719

### CODE REVIEW

File: osu.Game/IO/OsuCachedOnlineStore.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// See the LICENCE file in the repository root for full licence text.
+
+using System.IO;
+using osu.Framework.IO.Stores;
+
+namespace osu.Game.IO
+{
+    public abstract class OsuCachedOnlineStore : CachedOnlineStore
+    {
+        protected override string CachePath => Path.Combine(Path.GetTempPath(), "osu");
```
Comment (UselessToucan) [inline osu.Game/IO/OsuCachedOnlineStore.cs]: This is incorrect and should be osu! installation directory("C:\Users\TocoT\AppData\Roaming\osu\cache" on my machine), but I don't know how to get the correct path.

**Comment by Game4all** [comment]
This could probably be cached at `OsuGame` level as other online overlays could possibly benefit from this and share a single, same cache folder.

**Comment by peppy** [comment]
I disagree with caching this. It’s very rare a user will request. a preview twice.

**Comment by peppy** [comment]
Closing this one as unnecessary. It may be that we want to use caching for avatars / user covers but definitely won't be caching audio previews for the reason in my previous comment (very rare request).

**Pull request closed without merging**